src/requirements/__test__/requirement-graph-builder.test.ts

```diff
@@ -88,81 +100,73 @@ it('buildRequirementFulfillmentGraph phase 2-2 test', () => {
-    requirements,
-    userCourses: [CS3410, CS3420, MATH4710],
-    userChoiceOnFulfillmentStrategy: { 'CS3410/CS3420': [CS3410.courseId] },
-    userChoiceOnDoubleCountingElimination: { [MATH4710.uniqueId]: 'Probability' },
-    getAllCoursesThatCanPotentiallySatisfyRequirement,
-    allowDoubleCounting: () => false,
-  });
+  const graph = buildRequirementFulfillmentGraph(
+    {
+      requirements,
+      userCourses: [CS3410, CS3420, MATH4710],
+      userChoiceOnFulfillmentStrategy: { 'CS3410/CS3420': [CS3410.courseId] },
+      userChoiceOnDoubleCountingElimination: { [MATH4710.uniqueId]: 'Probability' },
+      getAllCoursesThatCanPotentiallySatisfyRequirement,
+      allowDoubleCounting: () => false,
+    },
+    /* keepCoursesWithoutDoubleCountingEliminationChoice */ true
+  );
 
   expect(graph.getConnectedCoursesFromRequirement('CS3410/CS3420')).toEqual([CS3410]);
   expect(graph.getConnectedCoursesFromRequirement('Probability')).toEqual([MATH4710]);
-  // We need a functional phase 3-2 to eliminate CS3410 here.
   expect(graph.getConnectedCoursesFromRequirement('Elective')).toEqual([CS3410, CS3420]);
 });
 
 it('buildRequirementFulfillmentGraph phase 3 test 2', () => {
-  const { requirementFulfillmentGraph: graph } = buildRequirementFulfillmentGraph({
-    requirements,
-    userCourses: [CS3410, CS3420, MATH4710],
-    userChoiceOnFulfillmentStrategy: { 'CS3410/CS3420': [CS3410.courseId] },
-    userChoiceOnDoubleCountingElimination: { [MATH4710.uniqueId]: 'Elective' },
-    getAllCoursesThatCanPotentiallySatisfyRequirement,
-    allowDoubleCounting: () => false,
-  });
+  const graph = buildRequirementFulfillmentGraph(
+    {
+      requirements,
+      userCourses: [CS3410, CS3420, MATH4710],
+      userChoiceOnFulfillmentStrategy: { 'CS3410/CS3420': [CS3410.courseId] },
+      userChoiceOnDoubleCountingElimination: { [MATH4710.uniqueId]: 'Elective' },
+      getAllCoursesThatCanPotentiallySatisfyRequirement,
+      allowDoubleCounting: () => false,
+    },
+    /* keepCoursesWithoutDoubleCountingEliminationChoice */ true
+  );
 
   expect(graph.getConnectedCoursesFromRequirement('CS3410/CS3420')).toEqual([CS3410]);
   expect(graph.getConnectedCoursesFromRequirement('Probability')).toEqual([]);
-  // We need a functional phase 3-2 to eliminate CS3410 here.
   expect(graph.getConnectedCoursesFromRequirement('Elective')).toEqual([CS3410, CS3420, MATH4710]);
 });
 
-// The following 3 tests test that we will detect all double counted courses.
-it('buildRequirementFulfillmentGraph phase 4 test 1', () => {
-  const { illegallyDoubleCountedCourses } = buildRequirementFulfillmentGraph({
+// The following test ensures that we will remove edges when user makes no choice on courses.
+it('buildRequirementFulfillmentGraph phase 3 test 3', () => {
+  const graph = buildRequirementFulfillmentGraph({
     requirements,
     userCourses: [CS3410, CS3420, MATH4710],
     userChoiceOnFulfillmentStrategy: { 'CS3410/CS3420': [CS3410.courseId] },
```

Comment:
Does this mean that the course `CS3410` is applied to the `'CS3410/CS3420'` requirement or just that they're going to fulfill the requirement using CS 3410 but haven't applied to course to it yet?